### DISCUSSION TITLE
Use current site page title inside <title> tag

### DISCUSSION THREAD

**Discussion by alXDraghiciu** [OrchardCMS/OrchardCore#11117] [Q&A]
Hi.

In the Agency Theme or other example themes, inside the Layout.liquid file, the title tag is generated with the following:

`<title>{{ "PageTitle" | shape_new | shape_stringify }}</title>`

This will always result in the same title. 
The title of the current page is stored inside Model.ContentItem.DisplayText. However, that is not accessible inside Layout.liquid, only inside it's separate content item.

**What would be the way, Liquid-wise, to display the current page in the site, next to the tenant name?** Like _Home | MySite, Contact | MySite_. Adding code to Content.liquid also doesn't seem to do anything and manually changing every page type does work but seems tedious since we have multiple types of pages on the site.

**Comment by ns8482e** [comment]
Have you tried `page_title` filter ? https://docs.orchardcore.net/en/latest/docs/reference/modules/Liquid/#page_title

**Reply by alXDraghiciu** [reply]
It renders the Site.SiteName not the page name.

**Comment by Piedone** [comment]
The line you mention renders the title with the `PageTitle` shape, which in turn renders the title based on the "Page title format" under the admin -> Configuration -> General (if this is configured, otherwise will render a default title). This is Liquid and you can use the `page_title` filter indeed. E.g. this should produce what you're looking for:

```liquid
{% page_title Site.SiteName, position: "after", separator: " | " %}
```

**Reply by alXDraghiciu** [reply]
It does not work. It renders the same site name as before. And on nested pages it's followed by a separator and blank space.

**Reply by Piedone** [reply]
I tried before posting it here so it's supposed to work, at least in 1.2.2 and with a theme. Make sure that the current theme is indeed one that uses the `<title>{{ "PageTitle" | shape_new | shape_stringify }}</title>` code you've shown, like The Blog Theme (the Default theme is not suitable). (Since there's no title-related concept of nested pages in Orchard, multi-level titles you'll need to implement from Liquid yourself.)

**Reply by alXDraghiciu** [reply]
The current custom theme uses it. Maybe there's something wrong with how the pages are named? I use TitlePart on the pages and put the name there.

_It does seem to work, but only on nested pages._ For example I have a page named Stories containing multiple items with their own separate page. On Stories the page title is the site name instead of **Stories | Site name**, while on a nested page the title is Stories | Site name, not the **Story Title | Stories | Site name** as expected. So it seems to work but it's off by one level (I mean as in nesting level). 

Any ideas about that?

**Reply by Piedone** [reply]
How are you nesting these items, specifically?

**Reply by alXDraghiciu** [reply]
> How are you nesting these items, specifically?

They are contained in the Stories page and have their own AutoRoute part. So they are rendered in Stories as Summary view and at their own URL as Detail view.

**Reply by Piedone** [reply]
What do you mean that they're contained? Are you utilizing List Part, or Content Picker Field?

**Reply by alXDraghiciu** [reply]
The Story is a content item, with an AutoRoutePart. Then inside the Stories Page there's a named BagPart containing Story items. It's not using ListPart since that caused some issues.

**Reply by Piedone** [reply]
Hmm, I'm wondering how that works at all, since items in Bag Part are accessible individually (i.e. I don't know why you can open a Story by an Autoroute path).

In any case, by default, Title Part (or specifically, `ContentItem.DisplayText`) contributes to the page title only when the content items is displayed in the Detail display type. Is this what you display the items with?

**Reply by alXDraghiciu** [reply]
I'm not sure either but to my understanding ListPart behaves in an async way and that caused some issues if you try to reference it in another page? I'm not sure. BagPart on the other hand is sync and worked fine. It also made the content page less cluttered.

Yes I think they are displayed in Detail view because the template has no suffix so it's just Content-StoryPage.liquid for example. That results in Detail view by default right?

**Comment by deanmarcussen** [comment]
You are probably not rendering Model.Header from the bag part item in the detail view.

I’m guessing you’re only rendering Model.Content

should work if you do that, dd when I wrote contained routing

**Comment by alXDraghiciu** [comment]
Ok so the implementation you guys provided works, I've seen the same template use in the Blog theme example and there it does render correctly ( Page - Site ). 

For our site however, the same code seems to always render the site name. Maybe someone can help troubleshoot this.

So yes I think the header zone is disabled in our website so we don't render Model.Header in the pages. Could that be an issue?

**Reply by Piedone** [reply]
Yes.
